src/list-inactive.ts

```diff
@@ -0,0 +1,186 @@
+`;
+};
+
+const noRecentUpdate = (metadata: StackMetadata): Promise<InactiveStatus | undefined> => {
+  const twoYearsAgo = new Date();
+  twoYearsAgo.setFullYear(twoYearsAgo.getFullYear() - 2);
+  const stackDate = (metadata.LastUpdatedTime ?? metadata.CreationTime) as Date;
+  const isOld = twoYearsAgo > stackDate;
+  return isOld ? Promise.resolve("NoRecentUpdate") : Promise.resolve(undefined);
+};
+
+const testInName = (metadata: StackMetadata): Promise<InactiveStatus | undefined> => {
+  const matches = metadata.StackName?.toLowerCase().includes("test");
+  return matches ? Promise.resolve("TestInName") : Promise.resolve(undefined);
+};
+
+// Lambda - lastModified :)
+const noRecentLambdaDeployment = async (
+  clients: Clients,
+  metadata: StackMetadata
+): Promise<InactiveStatus | undefined> => {
+  // get lambda and then query lastmodified
+
+  const lambda = Object.entries(metadata.Template.Resources).find(
+    ([, resource]) => resource.Type === "AWS::Lambda::Function"
+  );
+
+  if (!lambda) return undefined;
+
+  const [id] = lambda;
+
+  const describeStackCmd = new DescribeStackResourceCommand({ StackName: metadata.StackName, LogicalResourceId: id });
+  const describeStackOutput = await clients.cfnClient.send(describeStackCmd);
+  const lambdaId = describeStackOutput.StackResourceDetail?.PhysicalResourceId;
+
+  const getFunctionCmd = new GetFunctionCommand({ FunctionName: lambdaId });
+  const fn = await clients.lambdaClient.send(getFunctionCmd);
+  const lastModifiedStr = fn.Configuration?.LastModified;
+
+  if (!lastModifiedStr) return undefined;
+
+  const sixMonthsAgo = new Date();
+  sixMonthsAgo.setMonth(sixMonthsAgo.getMonth() - 6);
+
+  const lastModified = new Date(lastModifiedStr);
+
+  console.debug(
+    `stack:${
+      metadata.StackName ?? "unknown"
+    }; lambda:${id}; lastModified:${lastModified.toISOString()}; oneMonthAgo:${sixMonthsAgo.toISOString()}`
+  );
+
+  return lastModified < sixMonthsAgo ? "NoRecentLambdaDeployment" : undefined;
+};
+
+const noRecentAsgDeployment = async (
+  clients: Clients,
+  metadata: StackMetadata
+): Promise<InactiveStatus | undefined> => {
+  const asg = Object.entries(metadata.Template.Resources).find(
+    ([, resource]) => resource.Type === "AWS::AutoScaling::AutoScalingGroup"
+  );
+
+  if (!asg) return undefined;
+
+  const [id] = asg;
+
+  const describeStackCmd = new DescribeStackResourceCommand({ StackName: metadata.StackName, LogicalResourceId: id });
+  const describeStackOutput = await clients.cfnClient.send(describeStackCmd);
+  const asgId = describeStackOutput.StackResourceDetail?.PhysicalResourceId;
+  const describeScalingActivitiesCmd = new DescribeScalingActivitiesCommand({ AutoScalingGroupName: asgId });
+  const describeScalingActivitiesOutput = await clients.asgClient.send(describeScalingActivitiesCmd);
+  const activities = describeScalingActivitiesOutput.Activities;
```

Comment:
Based on [these docs](https://docs.aws.amazon.com/cli/latest/reference/autoscaling/describe-scaling-activities.html#:~:text=Describes%20scaling%20activity%2C%20which%20is,size%20or%20replacing%20an%20instance.), I wonder if this will also contain scaling events that are not related to a deployment (e.g. replacing an unhealthy instance or scaling on CPU usage)? I would guess that the latter is rare but the former may be reasonably likely for older services?

If so, perhaps we can filter based on the desired capacity being increased?